restapi/src/main/java/io/stargate/web/docsapi/service/query/FilterExpression.java

```diff
@@ -44,6 +50,114 @@
   @Value.Parameter
   public abstract BaseCondition getCondition();
 
+  /** @return Returns human-readable description of this expression. */
+  public String getDescription() {
+    BaseCondition condition = getCondition();
+    return getFilterPath().getPathString()
+        + " "
+        + condition.getFilterOperationCode()
+        + " "
+        + condition.getQueryValue();
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * <p>Tests all rows in this document, making sure that at least is one the filter path and
+   * fulfills the condition. If no row is on the filter path, returns true.
+   */
+  @Override
+  public boolean test(RawDocument rawDocument) {
+    boolean anyFailing = false;
+    for (Row row : rawDocument.rows()) {
+
+      // if row is not matching the filter path do nothing
+      if (!matchesFilterPath(row)) {
+        continue;
+      }
+
+      // do a test
+      boolean test = getCondition().test(row);
+
+      // if the condition is passed, whole document passes
+      // otherwise, continue in hope to find a row passes
+      if (test) {
+        return true;
+      } else {
+        anyFailing = true;
+      }
+    }
+
+    // reaching here means either that no row matches the filter path
+    // or we had failing rows
+    return !anyFailing;
```

Comment:
Would it be more consistent to test the condition against `null` in this case (since the property value is missing)?

Reply:
What use case would this fail into? this `NE` check? I'll have a look if we need it..

Reply:
Seems I still don't need this.

Reply:
That's fine. Resolving.